enhancements/centralized-manifest-openapi-generation.md

```diff
@@ -0,0 +1,83 @@
+- [ ] Design details are appropriately documented
+- [ ] Test plan is defined
+
+## Summary
+This document proposes migrating operator CRD manifests (and, more importantly, the work associated with updating them)
+to be colocated with the Golang types that define them in the [openshift/api](https://github.com/openshift/api) repository.
+
+## Motivation
+We have many components that consume custom resources, and almost all of those resources are based on Golang types defined in
+`openshift/api`. When those Golang types change, or CRD requirements change such as with [Kubernetes 1.16's
+structural schemas](https://kubernetes.io/blog/2019/06/20/crd-structural-schema/), these changes require matching updates
+to each operator repo. This is primarily evident in changes to OpenAPIV3 validation schemas, where it is not uncommon
+to make small changes to API markers which then need to be individually carried to each operator where they are generated
+into new manifest changes. In addition, maintainers currently may have different approaches to generating their manifests
+(using different tools and scripts) which can lead to discrepancies and unexpected behavior without support.
+
+By moving CRD manifests to `openshift/api`, we will colocate these types with their consumers and enable
+organization-wide changes to these manifests to be enabled at once, included with the corresponding changes to their types.
+We also will provide a single approach to maintaining the API/manifest relationship, specifically in regards to generating
+OpenAPIV3 validation schemas.
+
+### Goals
+
+1. Operator and API maintainers can make changes to APIs and their corresponding CRD manifests simultaneously
+2. Manifest changes can be easily read from `openshift/api` to the operators that consume them
+3. Unify organization schema generation under a single tool and approach for CRD schema generation
+
+### Non-goals
+
+- We will not be responsible for maintaining individual component APIs or manifests. Though we are providing a single method
+to organize their management, maintainers will still be responsible for changes related to only their operator.
+
+## Proposal
+
+CRD manifests from relevant repositories will be moved to the same package  directories in `openshift/api` as the types that define them.
+This provides the benefit of allowing these types to be imported with Go modules, as well as inheriting the existing OWNERS
+files for those packages.
+
+Then, makefile targets will be added to `openshift/api` based on the [existing build machinery targets](https://github.com/openshift/library-go/blob/27b03247913b2f6a0abcea95cb9b6c896adc2531/alpha-build-machinery/make/targets/openshift/crd-schema-gen.mk) that currently enables OpenAPI schema generation. This generator will then be run when any API changes require
+an update to CRD manifests, and manifests will be verified by CI, similar to how these changes are currently tested in many repos.
```

Comment:
indicate that we will enable this on a per API group level and state your intent of which file will maintain the list of CRD generated groups or directories.  I don't actually know which.